lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1974,14 +1974,15 @@ NvidiaMmaEncodingAttr::getRepOrderForOperand(int opIdx) const {
 
 SmallVector<int64_t>
 NvidiaMmaEncodingAttr::getRepForOperand(ArrayRef<int64_t> shape, int bitwidth,
-                                        int opIdx) const {
+                                        int kWidth, int opIdx) const {
   auto rank = shape.size();
   auto warpsPerCTA = getWarpsPerCTA();
 
   // {batch, m, n, k}
   // Hopper path never uses the n value, since this method is only invoked
   // for in-RF (dotOpEnc) operands, but WGMMA only supports in A to be in RF
-  SmallVector<int> shapePerWarp = {1, 16, 8, 4 * 64 / bitwidth};
+  SmallVector<int> shapePerWarp = {
+      1, 16, 8, isHopper() ? 4 * 2 * kWidth : 4 * 64 / bitwidth};
```

Comment:
I don't think this is correct for Hopper. The number of reps does not depend on `kWidth`, as `kWidth` is part of the size, and thus the tile.

Reply:
> I don't think this is correct for Hopper

Right

Reply:
`rep` specifies how many mma/wgmma instructions should be used

Reply:
well the idea is that for mmav3 the mma operation will have to take a dot_operand where kWidth and bitwidth match, so we want the layout to match this way. Like I said I think kWidth the wrong parameter here but that's why it feels like number of reps should depend on kWidth.
I'm happy to fix it a different way if you have a different suggestion. WDYT?

Reply:
> I don't think this is correct for Hopper. The number of reps does not depend on kWidth, as kWidth is part of the size, and thus the tile.

Can you explain this a bit more. I thought the number of repeats = total size / tile size for each dimension. Indeed this is what the function looks like it's computing to me.

Reply:
Yes, it is indeed that, I miswrote. For reference, see
https://github.com/triton-lang/triton/blob/5acbbb7d8d6991547470c6c67f2cbcd2a778e287/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv2.cpp#L552-L556
where `callMma` issues one mma instruction.

Then, the rest of the computation of `getElemesPerThread` for Ampere just computes how many elements are there in the `K` dimension and multiplies by it (perhaps broadcasting the layout). I think we should just be able to share the logic from Ampere for Hopper here, as the layouts are equivalent when it comes to these computations.

Reply:
> > I don't think this is correct for Hopper. The number of reps does not depend on kWidth, as kWidth is part of the size, and thus the tile.
> 
> Can you explain this a bit more. I thought the number of repeats = total size / tile size for each dimension. Indeed this is what the function looks like it's computing to me.

I meant does not depend on *bitwidth*